R/macros.R

```diff
@@ -0,0 +1,99 @@
+#' General Confluence macro builder for internal use
+#' @param type inline or block code style to be used for the HTML content
+#' @param name Confluence macro name
+#' @param parameters named list of optional macro parameters
+#' @param body optional \code{confl-ac-rich-text-body} content
+#' @return HTML
+#' @keywords internal
+conf_macro_generator <- function(type = c('inline', 'block'),
+                                 name, parameters, body) {
+  type <- match.arg(type)
+
+  macro <- switch(
+    type,
+    inline = '`',
+    block = '\n```{=html}\n')
+
+  macro <- glue('{macro}<ac:structured-macro ac:name="{name}">')
+
+  if (!missing(parameters)) {
+    for (parameter in names(parameters)) {
+      macro <- paste0(
+        macro,
+        glue('<ac:parameter ac:name="{parameter}">'),
+        parameters[[parameter]],
+        '</ac:parameter>')
+    }
+  }
+
+  if (!missing(body)) {
```

Comment:
This is just a matter of preference, but I prefer setting `NULL` as the default of optional arguments and check `is.null()`. Could you follow this way?

c.f. https://design.tidyverse.org/def-required.html